### PR TITLE
ENH remove backend checks for mpi

### PR DESCRIPTION
**Pull request by hndgzkn**
Simplifies code removing all `backend  == "mpi"`checks.

MPI backend is not clearly separated. If another backend is to be implemented in the future, it would be better to implement in separate classes to a common Interface instead of if checks.

### CODE REVIEW

File: dicodile/workers/dicod_worker.py [hunk 1]
```diff
@@ -84,7 +77,8 @@ def compute_z_hat(self):
 
         diverging = False
         if flags.INTERACTIVE_PROCESSES and self.n_workers == 1:
-            import ipdb; ipdb.set_trace()  # noqa: E702
+            import ipdb
+            ipdb.set_trace()  # noqa: E702
```
Comment (tomMoral) [inline dicodile/workers/dicod_worker.py:81]: In this case, you need to remove the `noqa`
```suggestion
            ipdb.set_trace() 
```

**Review by tomMoral — COMMENTED** [review]
LGTM! thx @hndgzkn

**Review by tomMoral — APPROVED** [review]

File: dicodile/workers/dicod_worker.py [hunk 2]
```diff
@@ -84,7 +77,8 @@ def compute_z_hat(self):
 
         diverging = False
         if flags.INTERACTIVE_PROCESSES and self.n_workers == 1:
-            import ipdb; ipdb.set_trace()  # noqa: E702
+            import ipdb
+            ipdb.set_trace()
```
Comment (tomMoral) [inline dicodile/workers/dicod_worker.py:81]:
```suggestion
            ipdb.set_trace()
```

File: dicodile/workers/dicod_worker.py (hunk 1 above)
Comment (hndgzkn) [reply dicodile/workers/dicod_worker.py]: Sorry, it's my emacs which auto formats.

**Pull request merged**